docs/source/oidc.rst

```diff
@@ -0,0 +1,53 @@
+Open ID Connect (OIDC)
+======================
+Support for Open ID Connect has been added in 1.4.2 of Janeway. This guide aims to assist you in setting up OIDC with Janeway. Janeway makes use of the (`Mozilla Django OIDC <https://github.com/mozilla/mozilla-django-oidc>`_) package.
+
+If you are upgrading from 1.4.1 to 1.4.2+ ensure you reinstall all requirements so that any required packages are installed.
+
+Settings
+--------
+The majority of changes required to get OIDC working in Janeway are in your settings files.
+
+By default OIDC is disabled in janeway_global_settings module. You can overwrite this in your local settings file with the following:
+
+::
+
+    # OIDC Settings
+    ENABLE_OIDC = True
+    OIDC_SERVICE_NAME = 'Your SSO Service Name'
+    OIDC_RP_CLIENT_ID = 'your-client-name'
+    OIDC_RP_CLIENT_SECRET = ''
+    OIDC_RP_SIGN_ALGO = 'RS256'
+    OIDC_OP_AUTHORIZATION_ENDPOINT = "https://auth.example.org/auth/realms/REALMNAME/protocol/openid-connect/auth"
+    OIDC_OP_TOKEN_ENDPOINT = "https://auth.example.org/auth/realms/REALMNAME/protocol/openid-connect/token"
+    OIDC_OP_USER_ENDPOINT = "https://auth.example.org/auth/realms/REALMNAME/protocol/openid-connect/userinfo"
+    OIDC_OP_JWKS_ENDPOINT = 'https://auth.example.org/auth/realms/REALMNAME/protocol/openid-connect/certs'
+    OIDC_LOGOUT_URL = 'https://auth.example.orgg/auth/realms/REALMNAME/protocol/openid-connect/logout'
+
+    if ENABLE_OIDC:
+        MIDDLEWARE_CLASSES += (
+            'mozilla_django_oidc.middleware.SessionRefresh',
+        )
+
+    AUTHENTICATION_BACKENDS = (
+        'django.contrib.auth.backends.ModelBackend',
+        'utils.oidc.JanewayOIDCAB',
+    )
+
+.. note:: All of the urls above are examples derived from a Keycloak instance. You can get the real URLs by using your service providers .well-known url which will look something like:     https://exmaple/org/.well-known/openid-configuration. You can get your Client ID and Secret from your service provider admin console or can request one from your technical staff. If you use RS256 for OIDC_RP_SIGN_ALGO you must also complete OIDC_OP_JWKS_ENDPOINT otherwise login will fail.
+
+New Accounts
+------------
+When a user without an account on Janeway logs in a new active account is generated for them with their Email address as the django username, first name and last name are also copied.
+
+If you want to stop OIDC from creating accounts you can add `OIDC_CREATE_USER = False` to your settings file.
```

Comment:
Just curious as to what would happen in Janeway if this is set to False and I log in with an account that does not exist in Janeway

Reply:
Dropped this line.